TreeMaker/src/TreeMaker.cc

```diff
@@ -48,10 +48,32 @@ TreeMaker::TreeMaker(const edm::ParameterSet& iConfig) :
 	treeName = iConfig.getParameter<string>("TreeName");
 	doLorentz = iConfig.getParameter<bool>("doLorentz");
 	sortBranches = iConfig.getParameter<bool>("sortBranches");
-	//loop over all var type names to initialize TreeObjects
-	set<string> nameSet;
+
+	// parse the TitleMap
 	stringstream skipMessage;
 	stringstream message;
+	set<string> titleSet;
+	vector<string> titleVector = iConfig.getParameter< vector<string> >("TitleMap");
+	message << "TitleMap:\n";
+	for(unsigned i = 0; i < titleVector.size(); i+=2){
```

Comment:
I would check for an odd-size vector before this, and probably just throw an exception in that case
(it would be a bit clearer then just getting an out of range exception from `vector.at()`)

Reply:
Done